batch/batches/Storage/Periodic/KAsyncStoragePeriodicExport.class.php

```diff
@@ -0,0 +1,267 @@
+<?php
+
+/**
+ * process remote storage files and export using file handlers as required
+ *
+ * @package Scheduler
+ * @subpackage Storage
+ */
+class KAsyncStoragePeriodicExport extends KPeriodicWorker
+{
+	const MAX_EXECUTION_TIME = 3600;        // can be exceeded by one file sync
+	const IDLE_SLEEP_INTERVAL = 10;
+	const MAX_COUNT = 'maxCount';
+	const MAX_SIZE = 'maxSize';
+	const STORAGE_PROFILE_IDS = 'profileIdsIn';
+	const RESPONSE_PROFILE_FIELDS = 'id,originalId,fileSize,fileRoot,filePath,isDir,srcPath,srcEncKey';
+
+	protected $storageProfiles;
+	protected $currentIndex;
+
+	/* (non-PHPdoc)
+	 * @see KBatchBase::getType()
+	 */
+	public static function getType()
+	{
+		return KalturaBatchJobType::STORAGE_PERIODIC_EXPORT;
+	}
+
+	/* (non-PHPdoc)
+	 * @see KBatchBase::run()
+	*/
+	public function run($jobs = null)
+	{
+		$filter = $this->getFilter();
+		$maxCount = $this->getAdditionalParams(self::MAX_COUNT);
+		$maxSize = $this->getAdditionalParams(self::MAX_SIZE);
+		$this->storageProfileIdsArray = explode(',', $this->getAdditionalParams(self::STORAGE_PROFILE_IDS));
+		if(!$this->storageProfileIdsArray)
+		{
+			KalturaLog::debug('Storage Profile Ids are missing from config');
+			return;
+		}
+		$this->currentIndex = 0;
+		$responseProfile = $this->initResponseProfile();
+		$timeLimit = time() + self::MAX_EXECUTION_TIME;
+
+		while (time() < $timeLimit)
+		{
+			$storageProfile = $this->getStorageProfile();
+			if(!$storageProfile)
+			{
+				continue;
+			}
+
+			self::$kClient->setResponseProfile($responseProfile);
```

Comment:
This can be done once outside the loop

Reply:
we cant extract it outside as there are other actions that use different response profile after we call this action so in the next iteration we must set the relevant response profile for the lock command